src/Enum.php

```diff
@@ -209,9 +209,8 @@ public static function toArray()
      * Check if is valid enum value
      *
      * @param $value
-     * @psalm-param mixed $value
+     * @psalm-param T $value
```

Comment:
imo this doesn't make sense, if i declare value as T i would already knew it is valid, no?

Maybe you can rather prepare a fail case which proofs that there is a problem with the current implementation?

Reply:
Let's assume this class:

```
/**
 * @extends Enum<string>
 */
class Status extends Enum {
    const SUCCESS = 'SUCCESS';
    const FAILURE = 'FAILURE';
}
```

The behavior of isValid is as follows:

```php
var_dump(Status::isValid('SUCCESS')); // true
var_dump(Status::isValid('UNKNOWN')); // false
var_dump(Status::isValid(1)); // false
```

With your implementation static analysis fails for all 3:

```php
// Call to static method MyCLabs\Enum\Enum<string>::isValid() with 'SUCCESS' will always evaluate to true. 
var_dump(Status::isValid('SUCCESS')); // true

// Call to static method MyCLabs\Enum\Enum<string>::isValid() with 'UNKNOWN' will always evaluate to true. 
var_dump(Status::isValid('UNKNOWN')); // false + the static analysis is incorrect

// Call to static method MyCLabs\Enum\Enum<string>::isValid() with 1 will always evaluate to false. 
var_dump(Status::isValid(1)); // false
```

With my suggestion:

```php
var_dump(Status::isValid('SUCCESS')); // true, static analysis passes
var_dump(Status::isValid('UNKNOWN')); // false, static analysis passes
var_dump(Status::isValid(1)); // false, static analysis fails saying that isValid expects a string, int given
```

Reply:
imo the problem is your `Enum<string>` annotation, with it you declare T as any string, but it is only `SUCCESS` and `FAILURE`.

Can your try this:

```php
/**
 * @extends Enum<Status::*>
 */
class Status extends Enum {
    const SUCCESS = 'SUCCESS';
    const FAILURE = 'FAILURE';
}
```

Reply:
Ooooh! Okay, interesting. I was completely unaware of that syntax. With that it works fine of course. It should be mentioned in the readme somewhere. Thanks!

Reply:
Agree with the readme, maybe you can just open a PR for it?

In addition, worth to mention you can also use only a specific scope of constants

```php
/**
 * @extends Enum<Status::VALUE_*>
 */
class Status extends Enum {
    const VALUE_SUCCESS = 'SUCCESS';
    const VALUE_FAILURE = 'FAILURE';
    const FOO = 'BAR';
}
``` 

or as literal values

```php
/**
 * @extends Enum<'SUCCESS'|'FAILURE'>
 */
class Status extends Enum {
    const SUCCESS = 'SUCCESS';
    const FAILURE = 'FAILURE';
}
```